### PR TITLE
Refine the uinput implementation

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- input/uinput: Rename UINPUT_TOUCHSCREEN to UINPUT_TOUCH 
- input: Move uinput initialization from boards/sim to drivers
- input/uinput: Remove the argument from initialization function 
- input/touchscreen: Change the buffer number type from uint8_t to uint32_t
- input/keyboard: Make each instance could have a different buffer size
- drivers: Call uinput_keyboard_initialize in drivers_initialize 
- boards/sim: Enable uinput for rpproxy and rpserver

## Impact
Minor, code refactor

## Testing
Pass CI and run with sim:rpproxy and sim:rpserver

### CODE REVIEW

File: drivers/input/Kconfig [hunk 1]
```diff
@@ -64,13 +54,25 @@ config UINPUT_RPMSG
 	---help---
 		Enable support uinput cross core communication
 
-config UINPUT_TOUCHSCREEN
-	bool "Enable uinput touchscreen"
+config UINPUT_TOUCH
```
Comment (pkarashchenko) [inline drivers/input/Kconfig:57]: Probably need to mark this with "breaking change" lable

File: drivers/input/keyboard_upper.c [hunk 2]
```diff
@@ -60,6 +60,7 @@ struct keyboard_upperhalf_s
   struct list_node head;    /* Head of list */
   FAR struct keyboard_lowerhalf_s
                   *lower;   /* A pointer of lower half instance */
+  uint32_t         nums;    /* Number of buffer */
```
Comment (pkarashchenko) [inline drivers/input/keyboard_upper.c:63]:
```suggestion
  size_t           nums;    /* Number of buffer */
```
Or maybe we can reduce it to `uint16_t`?

File: drivers/input/uinput.c [hunk 3]
```diff
@@ -453,19 +457,17 @@ static ssize_t uinput_keyboard_write(FAR struct keyboard_lowerhalf_s *lower,
...
  *
  ****************************************************************************/
 
-#ifdef CONFIG_UINPUT_TOUCHSCREEN
+#ifdef CONFIG_UINPUT_TOUCH
 
-int uinput_touch_initialize(FAR const char *name, int maxpoint, int buffnums)
+int uinput_touch_initialize(void)
 {
   char devname[UINPUT_NAME_SIZE];
```
Comment (pkarashchenko) [inline drivers/input/uinput.c:472]:
```suggestion
```

File: drivers/input/uinput.c [hunk 4]
```diff
@@ -41,8 +41,12 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-#define UINPUT_NAME_SIZE  16
-#define RPMSG_UINPUT_NAME "rpmsg-uinput-%s"
+#define UINPUT_NAME_SIZE     16
```
Comment (pkarashchenko) [inline drivers/input/uinput.c:44]:
```suggestion
```

File: drivers/input/uinput.c [hunk 5]
```diff
@@ -519,7 +522,7 @@ int uinput_touch_initialize(FAR const char *name, int maxpoint, int buffnums)
 
 #ifdef CONFIG_UINPUT_BUTTONS
 
-int uinput_button_initialize(FAR const char *name)
+int uinput_button_initialize(void)
 {
   char devname[UINPUT_NAME_SIZE];
```
Comment (pkarashchenko) [inline drivers/input/uinput.c:527]:
```suggestion
```

File: drivers/input/uinput.c [hunk 6]
```diff
@@ -569,7 +572,7 @@ int uinput_button_initialize(FAR const char *name)
 
 #ifdef CONFIG_UINPUT_KEYBOARD
 
-int uinput_keyboard_initialize(FAR const char *name)
+int uinput_keyboard_initialize(void)
 {
   char devname[UINPUT_NAME_SIZE];
```
Comment (pkarashchenko) [inline drivers/input/uinput.c:577]:
```suggestion
```

File: drivers/input/uinput.c [hunk 7]
```diff
@@ -478,29 +480,30 @@ int uinput_touch_initialize(FAR const char *name, int maxpoint, int buffnums)
...
   utcs_lower->ctx.notify     = uinput_touch_notify;
 #endif
 
   /* Regiest Touchscreen device */
 
-  snprintf(devname, sizeof(devname), "/dev/%s", name);
-  ret = touch_register(&utcs_lower->lower, devname, buffnums);
+  snprintf(devname, sizeof(devname), "/dev/%s", UINPUT_NAME_TOUCH);
+  ret = touch_register(&utcs_lower->lower, devname,
+                       CONFIG_UINPUT_TOUCH_BUFNUMBER);
```
Comment (pkarashchenko) [inline drivers/input/uinput.c:492]:
```suggestion
  ret = touch_register(&utcs_lower->lower,
                       "/dev/" UINPUT_NAME_TOUCH,
                       CONFIG_UINPUT_TOUCH_BUFNUMBER);
```

File: drivers/input/uinput.c [hunk 8]
```diff
@@ -534,7 +537,7 @@ int uinput_button_initialize(FAR const char *name)
   ubtn_lower->ctx.notify         = uinput_button_notify;
 #endif
 
-  snprintf(devname, sizeof(devname), "/dev/%s", name);
+  snprintf(devname, sizeof(devname), "/dev/%s", UINPUT_NAME_BUTTON);
   ret = btn_register(devname, &ubtn_lower->lower);
```
Comment (pkarashchenko) [inline drivers/input/uinput.c:541]:
```suggestion
  ret = btn_register("/dev/" UINPUT_NAME_BUTTON, &ubtn_lower->lower);
```

File: drivers/input/uinput.c [hunk 9]
```diff
@@ -589,8 +592,9 @@ int uinput_keyboard_initialize(FAR const char *name)
 
   /* Regiest Touchscreen device */
 
-  snprintf(devname, sizeof(devname), "/dev/%s", name);
-  ret = keyboard_register(&ukbd_lower->lower, devname);
+  snprintf(devname, sizeof(devname), "/dev/%s", UINPUT_NAME_KEYBOARD);
+  ret = keyboard_register(&ukbd_lower->lower, devname,
+                          CONFIG_UINPUT_KEYBOARD_BUFNUMBER);
```
Comment (pkarashchenko) [inline drivers/input/uinput.c:597]:
```suggestion
  ret = keyboard_register(&ukbd_lower->lower,
                          "/dev/" UINPUT_NAME_KEYBOARD,
                          CONFIG_UINPUT_KEYBOARD_BUFNUMBER);
```

File: drivers/input/Kconfig (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/input/Kconfig:57]: uinput doesn't officially release yet, so there isn't compatibility issue.

File: drivers/input/uinput.c (hunk 9 above)
Comment (xiaoxiang781216) [reply drivers/input/uinput.c:597]: Done.

File: drivers/input/uinput.c (hunk 8 above)
Comment (xiaoxiang781216) [reply drivers/input/uinput.c:541]: Done.

File: drivers/input/uinput.c (hunk 7 above)
Comment (xiaoxiang781216) [reply drivers/input/uinput.c:492]: Done.

File: drivers/input/uinput.c (hunk 6 above)
Comment (xiaoxiang781216) [reply drivers/input/uinput.c:577]: Done.

File: drivers/input/uinput.c (hunk 5 above)
Comment (xiaoxiang781216) [reply drivers/input/uinput.c:527]: Done.

File: drivers/input/uinput.c (hunk 4 above)
Comment (xiaoxiang781216) [reply drivers/input/uinput.c:44]: It's still used by uinput_context_s

File: drivers/input/uinput.c (hunk 3 above)
Comment (xiaoxiang781216) [reply drivers/input/uinput.c:472]: Done.

File: drivers/input/keyboard_upper.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/input/keyboard_upper.c]: Ok, I change to uint8_t

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**